### ISSUE TITLE
Lints don't work when a class's interface is spread out across multiple files

### ISSUE THREAD

**Issue by michaeleiselsc** [oclint/oclint#415]
### Issue Summary
When a method is declared with a modifier, `MissingCallToBaseMethod`, and it's in a category of the superclass that the subclass does not import, then it won't trigger any lints for that subclass, even if the file with the category is included in compile_commands.json

### Environment
- Operation system name and version: OS X 10.11.6, El Capitan
- OCLint version: 
```
LLVM (http://llvm.org/):
  LLVM version 3.9.1
  Optimized build.
  Default target: x86_64-apple-darwin15.6.0
  Host CPU: haswell

OCLint (http://oclint.org/):
  OCLint version 0.11.1.
  Built Jan  8 2017 (11:08:47).
```
- How OCLint is installed: homebrew

### Reproduction Steps
Use the files below to reproduce this issue

### Expected Behavior
The lint error should trigger with the files below

### Actual Behavior
The lint error does not trigger

### Even Better
See [Archive.zip](https://github.com/oclint/oclint/files/840070/Archive.zip) for an example

**Comment by AbigailBuccaneer** [comment]
This is the example from the archive:

```Objective-C
// Base.h
@interface Base
@end
```
```Objective-C
// Base.m
@interface Base (Category)
- (void)foo __attribute__((annotate("oclint:enforce[base method]")));
@end
@implementation Base (Category)
- (void)foo {}
@end
@implementation Base
@end
```
```Objective-C
// Derived.h
#import "Base.h"
@interface Derived : Base
@end
```
```Objective-C
// Derived.m
#import "Derived.h"
@implementation Derived
- (void)f {} // should trigger MissingCallToBaseMethod but doesn't
@end
```

I think this bug report is invalid - OCLint (just like the compiler) analyses files one translation unit at a time. When analysing Derived.m, OCLint has no way of knowing that some other translation unit is going to declare a category on `Base` with a method signature that matches one of `Derived`'s. As far as it's concerned, `Derived::f` isn't overriding anything.

I'd say that if there is an issue here, it's that OCLint doesn't do whole-program analysis and tell the programmer that `Derived` overrides methods of a base class' category defined elsewhere, or that `Base (Category)` has methods that conflict with derived classes elsewhere. And if that's fixed by making the category interface visible in `Derived.m`, *then* `MissingCallToBaseMethod` should give an error about `Derived::f`.

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.

**Comment by ryuichi-assistant** [comment]
This issue will be auto-closed because there hasn't been any activity for a few months. Feel free to open a new one if you still experience this problem.
